### PR TITLE
ARROW-5220: [Python] Follow-up to improve error messages and docs for from_pandas schema argument

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Some left-over TODO's from https://github.com/apache/arrow/pull/5379

### CODE REVIEW

File: python/pyarrow/pandas_compat.py [hunk 1]
```diff
@@ -412,14 +412,18 @@ def _get_columns_to_convert_given_schema(df, schema, preserve_index):
                 col = df.index.get_level_values(name)
                 if (preserve_index is None and
                         isinstance(col, _pandas_api.pd.RangeIndex)):
-                    # TODO better error message
-                    raise KeyError("name {} in schema not in columns "
-                                   "or index".format(name))
+                    raise ValueError(
+                        "name '{}' is present in the schema, but it is a "
```
Comment (bkietz) [inline python/pyarrow/pandas_compat.py]:
```suggestion
                        f"name '{name}' is present in the schema, but it is a "
                        "RangeIndex. Range indices are saved as metadata only- "
                        "they are not converted to a column. Specify 'preserve_index=True' "
                        "to force saving as a column or remove '{name}' from the specified schema."
```

**Review by wesm — APPROVED** [review]
+1

**Review by bkietz — CHANGES_REQUESTED** [review]

File: python/pyarrow/pandas_compat.py (hunk 1 above)
Comment (jorisvandenbossche) [reply python/pyarrow/pandas_compat.py]: That's unfortunately only a recent python feature .. ;) (even when dropping python 2.7, we would not yet be able to use it)

**Comment by wesm** [comment]
travis: https://travis-ci.org/jorisvandenbossche/arrow/builds/586655606
appveyor: https://ci.appveyor.com/project/jorisvandenbossche/arrow/builds/27503262

**Pull request closed without merging**